src/bun.js/webcore/streams.zig

```diff
@@ -3068,12 +3122,24 @@ pub fn ReadableStreamSource(
             }
         }
 
-        pub fn deinit(this: *This) void {
+        pub fn incrementCount(this: *This) !void {
             if (this.deinited) {
-                return;
+                return error.InvalidStream;
             }
-            this.deinited = true;
-            deinit_fn(&this.context);
+            this.ref_count += 1;
+        }
+
+        pub fn decrementCount(this: *This) u32 {
+            if (this.ref_count == 0 or this.deinited) {
+                return 0;
+            }
+
+            this.ref_count -= 1;
+            if (this.ref_count == 0) {
+                this.deinited = true;
+                deinit_fn(&this.context);
+            }
+            return this.ref_count;
```

Comment:
UAF

Reply:
fixed in https://github.com/oven-sh/bun/pull/4418/commits/7cd66913e79f7c27d13efeb30801dfcbbabe45e6